### ISSUE TITLE
File chooser dialog needs better screenreader support

### ISSUE THREAD

**Issue by Venefilyn** [cockpit-project/cockpit#23705]
Within the file chooser dialog the screenreader functionality for the elements, search, navigation, and more lack aria labels, have too many labels, or have strange structure to screenreader users.

We need to go through the elements with a screenreader using regular usage as well as keyboard usage to see how it sounds and make changes if it doesn't work out or otherwise is confusing.
